File: C,C++/p11413.c
#include<stdio.h>
#include<limits.h>

int banyak = 0, container = 0;

int countMax(int arr[banyak]) {
    int res = 0, i = 0;
    for (;i < banyak; i++) {
        res += arr[i];
    }
    
    return res;
}

int cekJawaban(int arr[banyak], int target, int targetCont) {
    int res = 0, cont = 1, i =0;
    for (;i < banyak; i++) {
        if ((res + arr[i]) > target) {
            res = arr[i];
            cont++;
            
            if (res > target) return 0;
        } else {
            res += arr[i];
        }
        
    }
    
    if (cont > targetCont) return 0;
    else return 1;
}

int search(int arr[banyak], int low, int high, int targetCont) {
    int res = INT_MAX;
    while(1) {
        if (high < low) return res;
        else {
            int mid = (high+low) / 2;
            if (cekJawaban(arr,mid,targetCont)) {
                if (mid < res) res = mid;
                high = mid-1;
            } else {
                low = mid+1;
            }
        }
    }
}

int main() {
   while (scanf("%d %d", &banyak, &container) == 2) {
       int arr[banyak];
       int i = 0;
       for (;i < banyak; i++) {
           scanf("%d", &arr[i]);
       }
       
       printf("%d\n", search(arr, 0, countMax(arr), container));
   }
   
   return 0;
}

File: Java/p11413.java
import java.util.Scanner;

public class FillTheContainers {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        // int kasus = input.nextInt();
        // for (int i = 0; i < kasus; i++) {
        
        while(input.hasNext()) {
            int banyak = input.nextInt();
            int container = input.nextInt();
            int [] arr = new int[banyak];
            for (int j = 0; j < banyak; j++) {
                arr[j] = input.nextInt();
            }
            
            System.out.println(search(arr, 0, countMax(arr), container));
        }
    }
    
    static int countMax (int [] arr) {
        int res = 0;
        for (int i = 0; i < arr.length; i++) {
            res+=arr[i];
        }
        return res;
    }
    
    static int search (int [] arr, int low, int high, int targetContainer) {
        int res = Integer.MAX_VALUE;
        while (true) {
            if (high < low) return res;
            else {
                int mid = (high+low) / 2;
                if (cekJawaban(arr, mid, targetContainer)) {
                    if (mid < res) res = mid;
                    high = mid-1;
                } else {
                    low = mid+1;
                }
            }
        }
    }
    
    static boolean cekJawaban(int [] arr, int target, int targetContainer) {
        int res = 0;
        int container = 1;
        for (int i = 0; i < arr.length; i++) {
            if ((res + arr[i]) > target) {
//                if (i != arr.length-1) {
                    res = arr[i];
                    container++;
//                }
                if (res > target) return false;
            } else {
                res += arr[i];
            }
        }
        if (container > targetContainer) return false;
        else return true;
    }
}